src/Navigation/Notifications/Repository/Notification.php

```diff
@@ -100,6 +100,36 @@ public function selectForUser(int $uid, array $condition = [], array $params = [
 		return $this->select($condition, $params);
 	}
 
+	/**
+	 * Returns only the most recent notifications for the same conversation or contact
+	 *
+	 * @param int $uid
+	 * @return Collection\Notifications
+	 * @throws Exception
+	 */
+	public function selectDigestForUser(int $uid): Collection\Notifications
+	{
+		$rows = $this->db->p("
```

Comment:
I haven't recognized this query when looking at the code. What is this query supposed to do, why is it so complicated?

Reply:
This is the query that determines the most recent notification for any given thread. How else would you do it?

Reply:
Again, how else would you do it? Not all notifications are pertaining to a thread, so the next grouping mechanism is by actor to get the most recent notification from the same contact. This is the purpose of the `IFNULL()` call, as a fallback when `parent-uri-id` isn't set.

Reply:
I have to make some tests to check which query is the best.

Reply:
I use a ``IN (SELECT MAX(`id`) GROUP BY)`` because MySQL's behavior with the order of the lines matched by a GROUP BY clause isn't defined. This is a consequence of allowing non-GROUP BY columns in the SELECT. To consistently retrieve the most recent row per matched group, you need to perform an aggregate function (here, `MAX()`) to obtain the id of the expected row, and then match on this id to retrieve the complete record.

I do not know how to do it differently in (My)SQL.

Reply:
I find having a single keyword on each line to be more readable, but it isn't the first time you've exhibited programming chaos-loving behavior!